### DISCUSSION TITLE
CIS-CAT

### DISCUSSION THREAD

**Discussion by v-stoichkov** [wazuh/wazuh#16035] [General]
Hello guys, 

I am trying to implement CIS CAT. I have license for CIS CAT Pro, but looks like I miss something, because Wazuh only using SCA so far.
So I installed CIS CAT Pro under `/var/ossec/wodles/ciscat` executable script is granted with +x and also is rename it from Assessor-CLI.sh to CIS-CAT.sh
In `ossec.conf` file i have following setting regarding cis-cat:

```
<wodle name="cis-cat">
    <disabled>no</disabled>
    <timeout>1800</timeout>
    <interval>10m</interval>
    <scan-on-start>yes</scan-on-start>

    <java_path>/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.161-2.b14.el7.x86_64/jre/bin/</java_path>
                <ciscat_path>/var/ossec/wodles/ciscat/</ciscat_path>
                <content type="xccdf" path="/var/ossec/wodles/ciscat/benchmarks/CIS_Microsoft_Windows_10_Enterprise_Benchmark_v1.12.0-xccdf.xml">
                        <profile>xccdf_org.cisecurity.benchmarks_profile_Level_1_L1_-_CorporateEnterprise_Environment_general_use</profile>
          </content>
  </wodle>
```
So I am trying to scan my Laptop which is running Windows 10

Is any one can help me with configuration?

**Comment by pereyra-m** [comment]
Hello @v-stoichkov !

Can you share your full `ossec.log` file? Don't forget to remove any sensitive information first.

If there is any problem with the module we should see a related log entry.
Also, what is your Wazuh version in both manager and agent?

Regards.

**Comment by v-stoichkov** [comment]
Hello @pereyra-m, 

Please check attached log files where I grep only 'wm_ciscat.c' related lines. If more of the logs is required will export the whole file
[ciscat_log.txt](https://github.com/wazuh/wazuh/files/10557618/ciscat_log.txt)

Regarding the versions, are as follow:
Wazuh agent v4.3.8
Wazuh manager v4.3.8

**Comment by pereyra-m** [comment]
Hello again @v-stoichkov!

Thank you very much for the logs provided.
The error messages like these might indicate that the scan wasn't run for some reason

```
DEBUG: Report result file 'tmp/ciscat-report.txt' missing: No such file or directory
ERROR: Failed reading scan results for policy '/var/ossec/wodles/ciscat/benchmarks/CIS_Microsoft_Windows_10_Enterprise_Benchmark_v1.12.0-xccdf.xml'
```

There is another issue related to the same logs https://github.com/wazuh/wazuh/issues/15453 and there it's explained that CIS-CAT requires a perfect match between the benchmark and the OS.
Can you confirm that the configured policy is correct for that agent? what is the exact OS version of the agent?

On the other side, have you followed the guide [CIS-CAT integration](https://documentation.wazuh.com/4.3/user-manual/capabilities/policy-monitoring/ciscat/ciscat.html#cis-cat-integration)? 
There we can see that CIS-CAT must be installed and configured in the agent, the manager only receives the alerts.
The configuration block you've shown has Windows policy but a JAVA path in Unix format, it seems to be applied in the manager.

Regards.
